napari/_qt/qt_viewer.py

```diff
@@ -370,33 +371,22 @@ def _create_canvas(self) -> None:
     def _diconnect_theme(self):
         self.viewer.events.theme.disconnect(self.canvas._on_theme_change)
 
-    def _add_visuals(self) -> None:
-        """Add visuals for axes, scale bar, and welcome text."""
+    def _add_overlay_visuals(self) -> None:
+        """Add visuals for overlays."""
+
+        for i, (name, overlay) in enumerate(self.viewer.overlays):
+            if name == 'visible':
+                continue
+            vispy_overlay = create_vispy_overlay(
+                overlay, viewer=self.viewer, parent=self.view, order=1e6 + i
```

Comment:
I think the problem might be that for the interaction_box the parent should be view.scene and not view. I guess that means it wants to live in world coordinates and not in canvas coordinates.